src/client/kube_store/index.ts

```diff
@@ -221,6 +35,58 @@ export default class KubeStore {
     return d['clusters'][this.clusterName];
   }
 
+  private determineKey(resource: KubeResource, name: string): object {
```

Comment:
Note this is different than what we discussed.

We discussed that we'd refactor out a method to determine the key, hence the name of this function is 'determineKey', yet this function as written now is returning the actual value and not the key.